### PR TITLE
HA additions

### PR DESCRIPTION
**Pull request by apdibbo**
Adds configuration to support a number of High Availability features including:
- Failover of l3 agents
- Multiple DHCP agents per subnet
- Haproxy configuration for Openstack components
- Support for multiple controller nodes 
- Memcache for all supported services

### CODE REVIEW

File: defaults/openstack/ha.pan [hunk 1]
```diff
@@ -0,0 +1,39 @@
+template defaults/openstack/ha;
+
+variable OS_GLANCE_PORT = 9292;
```
Comment (guillaume-philippon) [inline defaults/openstack/ha.pan]: I think it's a better thing to use ?= operator instead of = and keep variable definition on config template.

**Comment by jrha** [comment]
@apdibbo _ping_

**Comment by apdibbo** [comment]
I believe I have addressed the comments here

**Comment by guillaume-philippon** [comment]
Hi,

After merging mongodb PR this one is no more mergeable.

**Pull request merged**